### ISSUE TITLE
Failing to deploy k8s CronJobs

### ISSUE THREAD

**Issue by masonwr** [spinnaker/spinnaker#2792]
### Issue Summary:

Attempting to deploy a k8s CronJob results in the following exception:

```
Exception ( Monitor Deploy )

java.lang.NullPointerException at com.netflix.spinnaker.clouddriver.kubernetes.v2.op.manifest.KubernetesDeployManifestOperation.operate(KubernetesDeployManifestOperation.java:112) at com.netflix.spinnaker.clouddriver.kubernetes.v2.op.manifest.KubernetesDeployManifestOperation.operate(KubernetesDeployManifestOperation.java:51) at com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation$operate.call(Unknown Source) at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1$_closure2.doCall(DefaultOrchestrationProcessor.groovy:89) at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1$_closure2.doCall(DefaultOrchestrationProcessor.groovy) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93) at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325) at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294) at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022) at groovy.lang.Closure.call(Closure.java:414) at groovy.lang.Closure.call(Closure.java:408) at com.netflix.spinnaker.clouddriver.metrics.TimedCallable$ClosureWrapper.call(TimedCallable.groovy:55) at com.netflix.spinnaker.clouddriver.metrics.TimedCallable.call(TimedCallable.groovy:82) at java_util_concurrent_Callable$call$0.call(Unknown Source) at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1.doCall(DefaultOrchestrationProcessor.groovy:88) at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1.doCall(DefaultOrchestrationProcessor.groovy) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93) at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325) at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294) at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022) at groovy.lang.Closure.call(Closure.java:414) at groovy.lang.Closure.call(Closure.java:408) at com.netflix.spinnaker.security.AuthenticatedRequest.lambda$propagate$0(AuthenticatedRequest.java:59) at com.netflix.spinnaker.clouddriver.metrics.TimedCallable.call(TimedCallable.groovy:82) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:748)
```

### Cloud Provider(s):
Google Kubernetes Engine

### Environment:
Spinnaker 1.6.0 on GKE deploying to a different GKE

### Feature Area: 
provider/kubernetes-v2

### Steps to Reproduce:

Try to deploy:

``` yaml
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: simple-cron
spec:
  schedule: "*/2 * * * *"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
            - name: simple-cron-test
              image: busybox:latest
              command: ["echo"]
              args: ["hello", "world"]

          restartPolicy: Never
```

### Additional Details:

[Perhaps is related.](https://github.com/spinnaker/clouddriver/pull/2456)

**Comment by ethanfrogers** [comment]
@masonwr interestingly enough i was looking for this support this morning. that issue you mentioned is related. `CronJob` is not a supported resource kind.

**Comment by briananderson1222** [comment]
Was this supported in the v1 provider? Any thoughts on when this might become supported? Wouldn't a simple `apply` do the job here?

**Comment by lwander** [comment]
@nvndnl

**Comment by lwander** [comment]
Will be released in 1.8
